Is CMAKE >= 3.13 really necessary? Refuses to build on Ubuntu 18.04

I'm aware the CMake build was just recently added to the project. But requiring 3.13 is _very_ strict! Ubuntu 18.04, which is still in half its lifetime, ships with 3.10.2. Possibly the same happens in many other distros that are still supported.

Is the build script actually using any CMake features that are not available in 3.10? Is it possible for you guys to downgrade it, so it would build-able out-of-the-box by anyone? Sure, I can edit `CMakeLists.txt` myself, but that misses the point.

```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.13 or higher is required.  You are running version 3.10.2


-- Configuring incomplete, errors occurred!
```

@coldtobi : not sure if this will be very helpful, but here it goes:

```shell
03:38:17 rodrigo@desktop ~/install/timg/build main * $ cmake ../ -DWITH_VIDEO_DECODING=Off -DWITH_VIDEO_DEVICE=Off -DWITH_OPENSLIDE_SUPPORT=Off
-- The CXX compiler identification is GNU 7.5.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found Git: /usr/bin/git (found version "2.17.1") 
-- Checking for module 'GraphicsMagick++'
--   Found GraphicsMagick++, version 1.3.28
-- Checking for module 'libturbojpeg'
--   Found libturbojpeg, version 1.5.2
-- Checking for module 'zlib'
--   Found zlib, version 1.2.11
-- Checking for module 'libexif'
--   Found libexif, version 0.6.21
-- Checking for module 'libavutil'
--   Found libavutil, version 55.78.100
-- Checking for module 'libswscale'
--   Found libswscale, version 4.8.100
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
CMake Error at src/CMakeLists.txt:3 (add_executable):
  add_executable called with incorrect number of arguments


CMake Error at src/CMakeLists.txt:4 (target_sources):
  Cannot specify sources for target "timg" which is not built by this
  project.


CMake Error at src/CMakeLists.txt:25 (target_link_libraries):
  Cannot specify link libraries for target "timg" which is not built by this
  project.


-- Configuring incomplete, errors occurred!
See also "/home/rodrigo/install/timg/build/CMakeFiles/CMakeOutput.log".
See also "/home/rodrigo/install/timg/build/CMakeFiles/CMakeError.log".
```

The culprit, as I mentioned earlier, is the ` add_executable called with incorrect number of arguments`, as its API changed in 3.11, making the `sources` argument optional. They are declared in `target_sources()` (which I erroneously called `add_target()` before).

Maybe the solution is as simple as moving the sources declaration from `target_sources()` to `add_executable()`, so instead of:

```
add_executable(timg)
target_sources(timg PRIVATE
  timg.cc

  buffered-write-sequencer.h buffered-write-sequencer.cc
  display-options.h
  framebuffer.h     framebuffer.cc
  image-display.h   image-display.cc
  image-source.h    image-source.cc
  iterm2-canvas.h   iterm2-canvas.cc
  jpeg-source.h     jpeg-source.cc
  kitty-canvas.h    kitty-canvas.cc
  renderer.h        renderer.cc
  terminal-canvas.h terminal-canvas.cc
  termutils.h       termutils.cc
  thread-pool.h
  timg-base64.h
  timg-png.h        timg-png.cc
  timg-time.h
  unicode-block-canvas.h unicode-block-canvas.cc
)
```

We would have this:
```
add_executable(timg, timg.cc
  buffered-write-sequencer.h buffered-write-sequencer.cc
  display-options.h
  framebuffer.h     framebuffer.cc
  image-display.h   image-display.cc
  image-source.h    image-source.cc
  iterm2-canvas.h   iterm2-canvas.cc
  jpeg-source.h     jpeg-source.cc
  kitty-canvas.h    kitty-canvas.cc
  renderer.h        renderer.cc
  terminal-canvas.h terminal-canvas.cc
  termutils.h       termutils.cc
  thread-pool.h
  timg-base64.h
  timg-png.h        timg-png.cc
  timg-time.h
  unicode-block-canvas.h unicode-block-canvas.cc
)
```

But of course, I have no idea about the consequences of this blind change, not even sure if syntax is correct, or what becomes of `target_sources()` with this change

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.13)
+cmake_minimum_required(VERSION 3.10)
 
 project(timg VERSION 1.4.0 LANGUAGES CXX)
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,9 +1,7 @@
 set(CMAKE_CXX_FLAGS "-W -Wall -Wextra -Wno-unused-parameter -O3")
 
-add_executable(timg)
+add_executable(timg timg.cc)
 target_sources(timg PRIVATE
-  timg.cc
-
   buffered-write-sequencer.h buffered-write-sequencer.cc
   display-options.h
   framebuffer.h     framebuffer.cc

